### PR TITLE
Restrict protocol and ports definition for PacketCapture

### PR DESCRIPTION
**Pull request by asincu**
## Description
Restrict API definition for Ports to accept only numerical values. Named Ports will be rejected. Protocols will accept only full strings. 

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: pkg/apis/projectcalico/v3/packetcapture.go [hunk 1]
```diff
@@ -69,13 +69,12 @@ type PacketCaptureRule struct {
...
 	//
 	// Must be one of these string values: "TCP", "UDP", "ICMP", "ICMPv6", "SCTP", "UDPLite"
-	// or an integer in the range 1-255.
 	Protocol *numorstring.Protocol `json:"protocol,omitempty" validate:"omitempty"`
 
 	// Ports is an optional field that defines a filter for all traffic that has a
 	// source or destination port that matches one of these ranges/values. This value is a
 	// list of integers or strings that represent ranges of ports.
-	Ports []numorstring.Port `json:"ports,omitempty" validate:"omitempty,dive"`
+	Ports []numorstring.NumPort `json:"ports,omitempty" validate:"omitempty,dive"`
```
Comment (robbrockbank) [inline pkg/apis/projectcalico/v3/packetcapture.go:77]: One simpler option might be to just validate that the user has not set named ports here.  Then it feels a little more future-proofed (and doesn't require new types).

Comment (asincu) [reply pkg/apis/projectcalico/v3/packetcapture.go:77]: I was thinking about that. If we ever want to support this, then yes, I agree. If no, I think the API definition might be a better place than a runtime error.

Comment (asincu) [reply pkg/apis/projectcalico/v3/packetcapture.go:77]: I am doing the same for protocols, as in I created a validator to reject numerical values. I will simply update the descriptions on the fields to reflect this then.

File: pkg/apis/projectcalico/v3/packetcapture.go [hunk 2]
```diff
@@ -58,6 +58,8 @@ type PacketCaptureSpec struct {
 	// 	! has(label_name)
 	Selector string `json:"selector,omitempty" validate:"selector"`
 
+	// +kubebuilder:validation:MaxItems:=65535
```
Comment (robbrockbank) [inline pkg/apis/projectcalico/v3/packetcapture.go:61]: If these aren't doing anything I'd be inclined to remove them.

Did you see if this produced anything on the api server side?

**Review by robbrockbank — APPROVED** [review]
lgtm - one minor comment - I'm not too fussed either way, but it feels a little odd having directives that aren't being used at all.

File: pkg/apis/projectcalico/v3/packetcapture.go (hunk 2 above)
Comment (asincu) [reply pkg/apis/projectcalico/v3/packetcapture.go:61]: I am waiting on the https://github.com/tigera/apiserver/pull/464 to be merged so I can checked that.

Comment (freecaykes) [reply pkg/apis/projectcalico/v3/packetcapture.go:61]: @asincu fyi it's being moved to https://github.com/tigera/apiserver/pull/465

**Pull request closed without merging**